### PR TITLE
executor: handle multi column index.

### PR DESCRIPTION
**Pull request by coocood**
Multi column index scan was handled in `IndexScan` plan but not handled in `IndexScanExec` executor.

### CODE REVIEW

File: executor/executor.go [hunk 1]
```diff
@@ -234,8 +236,23 @@ func (e *IndexRangeExec) Next() (*Row, error) {
 	}
 }
 
+// indexCompare compares multi column index.
+// The length of boundVals may be less than idxKey.
+func indexCompare(idxKey []interface{}, boundVals []interface{}) (int, error) {
```
Comment (siddontang) [inline executor/executor.go]: I prefer
indexCompare -> indicesCompare 
indexSingleCompare -> indexCompare

Comment (coocood) [inline executor/executor.go]: Actually, the `indexSingleCompare` is not the whole index, but a single column in an index.

Comment (coocood) [inline executor/executor.go]: rename it to `indexColumnCompare` should be more accurate.

Comment (siddontang) [inline executor/executor.go]: yep

File: session_test.go [hunk 2]
```diff
@@ -1323,3 +1323,18 @@ func (s *testSessionSuite) TestErrorRollback(c *C) {
...
+	store := newStore(c, s.dbName)
+	se := newSession(c, store, s.dbName)
+	mustExecSQL(c, se, "drop table if exists t;")
+	mustExecSQL(c, se, "create table t (c1 int, c2 int);")
+	mustExecSQL(c, se, "create index idx_c1_c2 on t (c1, c2)")
+	mustExecSQL(c, se, "insert into t values (1, 5)")
+	mustExecMatch(c, se, "select c1 from t where c1 in (1) and c2 < 10", [][]interface{}{{1}})
+	mustExecMatch(c, se, "select c1 from t where c1 in (1) and c2 > 3", [][]interface{}{{1}})
+	mustExecMatch(c, se, "select c1 from t where c1 in (1.1) and c2 > 3", [][]interface{}{})
+	mustExecMatch(c, se, "select c1 from t where c1 in (1) and c2 < 5.1", [][]interface{}{{1}})
```
Comment (shenli) [inline session_test.go]: Any assertion to make sure that the executing process use index?

Comment (siddontang) [inline session_test.go]: maybe using a hook for test?

**Pull request merged**